durexforth.asm

```diff
@@ -97,13 +97,22 @@ _START = * + 1
 
 ; ----------- macros
 
-!set LINK = 0
-
-!macro BACKLINK {
-    ; it's tempting to add the string as a macro parameter,
-    ; but this does not seem to be supported by ACME.
-    !word	LINK
-    !set	LINK = * - 2
+!set TOP = $6fff
+!set DICTOP = TOP - 2
+!set BACK = *
+* = DICTOP
+!word 0
+!byte 0 ; we need this byte here to mark TOP without slowing FIND-NAME
+* = BACK
+
+!macro BACKLINK .name , .namesize {
+    !set DICTOP = DICTOP - 3 - len(.name)
+    !set .xt = *
+    * = DICTOP
+    !word .xt
+    !byte .namesize
+    !text .name

```

Comment:
I think it would make sense to change to

    .flags/namesize
    .name
    .xt

The most important operation on the wordlist is word finding. Then it becomes easier to understand if data is accessed by the order it is accessed. First flag byte, then name, then finally xt. This will also allow for having a single sentinel 0 byte at dictionary top.

Reply:
Hilarious.  That was my first design, but I switched it around for ease of finding XTs.

I'll find the commit and switch it back around.

Reply:
I'm not sure what you mean with ease of finding xt's..?
"FIND wordname" should return the xt ... but I'm probably missing something :-)